File: tfg/tfg_pagao/src/main/java/org/marino/server/data/models/entities/ParticipationEntity.java
package org.marino.server.data.models.entities;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "participation")
@IdClass(ParticipationEntityId.class)
public class ParticipationEntity {

    @Id
    @ManyToOne
    @JoinColumn(name = "member_id", referencedColumnName = "id")
    private MemberEntity member;

    @Id
    @ManyToOne
    @JoinColumn(name = "receipt_id", referencedColumnName = "id")
    private ReceiptEntity receipt;

    @Column(name = "description")
    private String description;

    @Column(name = "pays")
    private Double pays;

    @Column(name = "debts")
    private Double debts;
}

File: tfg/app/app/src/main/java/org/marino/tfgpagao/data/model/ParticipationEntity.kt
package org.marino.tfgpagao.data.model

import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(
    foreignKeys = [
        ForeignKey(
            entity = MemberEntity::class,
            parentColumns = ["id"],
            childColumns = ["memberId"],

            ),
        ForeignKey(
            entity = ReceiptEntity::class,
            parentColumns = ["id"],
            childColumns = ["receiptId"],
        )
    ],
    primaryKeys = ["memberId", "receiptId"],
    tableName = "participation"
)
data class ParticipationEntity (
    val memberId: Int,
    val receiptId: Int,
    val description: String?,
    val pays: Double?,
    val debts: Double?,
)